### ISSUE TITLE
~/.mill/ammonite/predef.sc is described in documentation but not used

### ISSUE THREAD

**Issue by hypatian** [lihaoyi/mill#924]
This is either a documentation error or a logic error. `~/.mill/ammonite/predef.sc` does not seem to be used or even touched by either `mill` or `mill -i`. Tested on both Mac OS and WSL2. This file is described in the documentation as being used for interactive mode, while `~/.mill/ammonite/predefScript.sc` is used only for command-line mode. http://www.lihaoyi.com/mill/page/configuring-mill.html#global-configuration

(As described in #820, `predefScript.sc` also fails, but for different reasons—and it causes failures for both interactive and command-line modes.)

**Comment by hypatian** [comment]
Okay, it turns out the documentation was just a little confusing. `predefScript.sc` is used when you're passing targets to the Mill command line, whether or not you use the `-i` switch. `predef.sc` is used and `predefScript.sc` is not when you pass the `-i` switch without providing a target to get an actual interactive session.

So, this is likely purely a documentation problem. Some places seem to use "interactive mode" to refer to any situation involving the `-i` switch to prevent a server from running:

>In some environments (such as WSL), mill might have to be run using interactive mode (-i)

and in fact in the `--help` output:
```
  -i, --interactive    Run Mill in interactive mode, suitable for opening REPLs and taking user input. In this mode, no mill server will be used.
```
but in this case, the actual distinction is between starting a Mill repl or providing a target (whether or not `-i` is used):
> Depending on how you start mill 2 different files will be loaded. For interactive mode it's `~/.mill/ammonite/predef.sc` and from the command line it's `~/.mill/ammonite/predefScript.sc`.

I'll make a PR to clarify this paragraph.

**Comment by lefou** [comment]
I guess, we need to introduce a dedicated cmdline option to disable mill-server, regardless of starting a Repl or not.

**Comment by hypatian** [comment]
That would definitely make the distinction between repl, target-with-no-server, and target-with-server a lot more clear.

**Comment by lihaoyi** [comment]
i think few enough people use the REPL that putting it behind a dedicated --repl flag would be best

**Comment by lefou** [comment]
See #926

**Comment by hypatian** [comment]
With the fix for issue #926 the documentation makes which predef file will be used more clear. Not a problem any more!
